app/src/renderer/vuex/modules/blockchain.js

```diff
@@ -33,36 +35,39 @@ export default ({ node }) => {
       dispatch(`subscribeToBlocks`)
     },
     async queryBlockInfo({ state, commit }, height) {
-      if (!height) {
-        commit(`notifyError`, {
-          title: `Couldn't query block`,
-          body: `No Height Provided`
+      try {
+        let blockMetaInfo = state.blockMetas[height]
+        if (blockMetaInfo) {
+          return blockMetaInfo
+        }
+        state.loading = true
```

Comment:
should we set `state.loading` outside of the try?

Reply:
it doesn't start loading anything prior